lib/services.js

```diff
@@ -27,41 +27,50 @@ async function startWebInspectorService (udid, opts = {}) {
   if (!semverVersion) {
     throw new Error(`Could not create a semver version out of '${osVersion}'`);
   }
-  const socketClient = opts.socket || await startService(udid, WEB_INSPECTOR_SERVICE_NAME);
-  return new WebInspectorService({
+  if (opts.socket) {
+    return new WebInspectorService({
+      majorOsVersion: semverVersion.major,
+      isSimulator,
+      socketChunkSize,
+      verbose,
+      verboseHexDump,
+      socketClient: opts.socket,
+    });
+  }
+  return await startService(udid, WEB_INSPECTOR_SERVICE_NAME, undefined, (socketClient) => new WebInspectorService({
     majorOsVersion: semverVersion.major,
     isSimulator,
     socketChunkSize,
     verbose,
     verboseHexDump,
     socketClient,
-  });
+  }));
 }
 
 async function startInstallationProxyService (udid, opts = {}) {
-  return new InstallationProxyService(new PlistService(await startService(udid, INSTALLATION_PROXY_SERVICE_NAME, opts.socket)));
+  return await startService(udid, INSTALLATION_PROXY_SERVICE_NAME, opts.socket, (socket) => new InstallationProxyService(new PlistService(socket)));
 }
 
 async function startAfcService (udid, opts = {}) {
-  return new AfcService(await startService(udid, AFC_SERVICE_NAME, opts.socket));
+  return await startService(udid, AFC_SERVICE_NAME, opts.socket, (socket) => new AfcService(socket));
 }
 
 async function startNotificationProxyService (udid, opts = {}) {
-  return new NotificationProxyService(await startService(udid, NOTIFICATION_PROXY_SERVICE_NAME, opts.socket));
+  return await startService(udid, NOTIFICATION_PROXY_SERVICE_NAME, opts.socket, (socket) => new NotificationProxyService(socket));
 }
 
 async function startHouseArrestService (udid, opts = {}) {
-  return new HouseArrestService(await startService(udid, HOUSE_ARREST_SERVICE_NAME, opts.socket));
+  return await startService(udid, HOUSE_ARREST_SERVICE_NAME, opts.socket, (socket) => new HouseArrestService(socket));
 }
 
-async function startService (udid, serviceName, socket) {
+async function startService (udid, serviceName, socket, connectCallback) {
```

Comment:
how about naming the callback `onConnect` and assigning it to null by default?

Reply:
same below

Reply:
Btw, you can send suggested commits on github comments. It is really cool feature

Reply:
@mykola-mokhnach `assigning it to null by default`? Why do you want that? Do you think `undefined` is worse than `null`?

Reply:
this is just to reflect that the argument is not mandatory

Reply:
I think I should then just add jsdoc. I added optional callbacks in the jsdoc of usbmux package